### PR TITLE
fixes issue #9 by hiding files with leading ._

### PR DESCRIPTION
**Pull request by seanjensengrey**

### CODE REVIEW

File: orion-viewer/src/main/java/universe/constellation/orion/viewer/FileChooser.java [hunk 1]
```diff
@@ -46,18 +46,35 @@
...
 
+    private static Set<String> extensions = new HashSet<>(Arrays.asList("cbz","djvu","djv","pdf","oxps","tif","tiff","xps"));
+
+
     public static FilenameFilter DEFAULT_FILTER = new FilenameFilter() {
-        public boolean accept(File dir, String filename) {
-        if (new File(dir, filename).isDirectory()) {
-            return true;
+        private String getExtension(String filename) {
+            for (int i = filename.length() - 1; i >= 0; i--) {
```
Comment (max-kammerer) [inline orion-viewer/src/main/java/universe/constellation/orion/viewer/FileChooser.java]: 'filename.substring(filename.lastIndexOf('.') + 1)' would be best

File: orion-viewer/src/main/java/universe/constellation/orion/viewer/FileChooser.java [hunk 2]
```diff
@@ -46,18 +46,35 @@
...
-        String name = filename.toLowerCase();
-        return name.endsWith(".pdf") || name.endsWith(".djvu") || name.endsWith(".djv") || name.endsWith(".xps") || name.endsWith(".oxps") || name.endsWith(".cbz")
-                || name.endsWith(".tiff") || name.endsWith(".tif") /*|| name.endsWith(".png") || name.endsWith(".jpeg") || name.endsWith(".jpg")*/;
+
+        public boolean accept(File dir, String filename) {
+            if (new File(dir, filename).isDirectory()) {
+                return true;
+            }
+            String name = filename.toLowerCase();
+            if (name.startsWith("._")) {
```
Comment (max-kammerer) [inline orion-viewer/src/main/java/universe/constellation/orion/viewer/FileChooser.java]: '_' is redundant, '.' would be enough

File: orion-viewer/src/main/java/universe/constellation/orion/viewer/FileChooser.java [hunk 3]
```diff
@@ -46,18 +46,35 @@
...
         }
-        String name = filename.toLowerCase();
-        return name.endsWith(".pdf") || name.endsWith(".djvu") || name.endsWith(".djv") || name.endsWith(".xps") || name.endsWith(".oxps") || name.endsWith(".cbz")
-                || name.endsWith(".tiff") || name.endsWith(".tif") /*|| name.endsWith(".png") || name.endsWith(".jpeg") || name.endsWith(".jpg")*/;
+
+        public boolean accept(File dir, String filename) {
+            if (new File(dir, filename).isDirectory()) {
+                return true;
+            }
+            String name = filename.toLowerCase();
```
Comment (max-kammerer) [inline orion-viewer/src/main/java/universe/constellation/orion/viewer/FileChooser.java]: It's non necessary to call 'toLowerCase' on filename, call it on extension would be enough

File: orion-viewer/src/main/java/universe/constellation/orion/viewer/FileChooser.java [hunk 4]
```diff
@@ -46,18 +46,35 @@
 
     private FilenameFilter filter;
 
+    private static Set<String> extensions = new HashSet<>(Arrays.asList("cbz","djvu","djv","pdf","oxps","tif","tiff","xps"));
```
Comment (max-kammerer) [inline orion-viewer/src/main/java/universe/constellation/orion/viewer/FileChooser.java]: Format line

**Comment by max-kammerer** [comment]
Please, squash commits and rebase them on current master

**Comment by seanjensengrey** [comment]
no problem, will have to wait a bit

File: orion-viewer/src/main/java/universe/constellation/orion/viewer/FileChooser.java [hunk 5]
```diff
@@ -46,14 +46,30 @@
...
         }
-        String name = filename.toLowerCase();
-        return name.endsWith(".pdf") || name.endsWith(".djvu") || name.endsWith(".djv") || name.endsWith(".xps") || name.endsWith(".oxps") || name.endsWith(".cbz")
-                || name.endsWith(".tiff") || name.endsWith(".tif") /*|| name.endsWith(".png") || name.endsWith(".jpeg") || name.endsWith(".jpg")*/;
+
+        public boolean accept(File dir, String filename) {
+            if (new File(dir, filename).isDirectory()) {
+                return true;
+            }
+            if (filename.startsWith("._") || filename.equals(".DS_Store")) {
```
Comment (max-kammerer) [inline orion-viewer/src/main/java/universe/constellation/orion/viewer/FileChooser.java]: Why do you need to filter DS_STORE files? They would be filtered by supported extensions below.

Comment (seanjensengrey) [inline orion-viewer/src/main/java/universe/constellation/orion/viewer/FileChooser.java]: The basic structure is

```
// allow all dirs
// deny filter
// allow filter (currently by file extension)
```

I understand that `.DS_Store` wouldn't make it through the allow filter, but it is more obvious that is in the set of things not shown.

**Comment by max-kammerer** [comment]
Many thanks!

**Pull request merged**